### PR TITLE
docs: add documentation for model caching on RunPod

### PR DESCRIPTION
**Pull request by rachfop**

### CODE REVIEW

**Review by maglietti — CHANGES_REQUESTED** [review]
The doc is missing instructions about how to access the cached models from inside the container. 

Lets. collaborate on this one.

**Review by Yhlong00 — APPROVED** [review]

**Comment by Yhlong00** [comment]
1. I was under the impression that model caching was specific to vLLM?
2. If users are working with a network volume, would model caching still be necessary? Or config model cache can cache the model on GPU server?
3. It might be helpful to provide an example of a model that uses a specific branch.

File: docs/pods/storage/model-caching.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- `HUGGINGFACE_TOKEN`: Token for accessing private models or datasets.
+- `HUGGINGFACE_USER`: Username for accessing private models or datasets.
+
+Examples:
+```
+HUGGINGFACE_MODEL="openchat/openchat-3.5-0106,openchat/openchat-3.5-1210"
+HUGGINGFACE_DATASET="google-bert/bert-base-uncased"
+```
+
+## Using Network Volumes
```
Comment (TimPietruskyRunPod) [inline docs/pods/storage/model-caching.md:34]: all of this has nothing to do with the modelcache, as it behaves out of the users scope. no network volume is needed.

instead we would need to know where the files are actually available after the container is starting:

* modelcache downloads the models / datasets from huggingface
* only when this is done, the pod / worker can actually start and execute what ever their purpose is (like running ComfyUI or doing inference for an LLM)
* the model is mounted read-only in `/runpod/cache/model/<organization>/<repo>/<branch>`
* users can then use the model / dataset inside their project and don't have to bake models into docker images anymore
* public models are shared with all users in the same datacenter
* private models can only be accessed by users who have access to the repo using their `RUNPOD_HUGGINGFACE_USER` + RUNPOD_HUGGINGFACE_TOKEN`. we always check the auth when users try to access this model. the model can then be found at `/runpod/cache/model/<user>/<organization>/<repo>/<branch>`

**Review by TimPietruskyRunPod — CHANGES_REQUESTED** [review]
There are some parts missing related how the modelcache works. I tried to explain everything, please let me know if something is not clear.

**Comment by TimPietruskyRunPod** [comment]
@Yhlong00 

> 1. I was under the impression that model caching was specific to vLLM?
nope, works with anything. 

> 2. If users are working with a network volume, would model caching still be necessary? Or config model cache can cache the model on GPU server?
yes, because you are not forced to stay in a specific datacenter, as you are not bound to a network volume. 

> 3. It might be helpful to provide an example of a model that uses a specific branch.
totally: `RUNPOD_HUGGINGFACE_MODEL`: `stabilityai/stable-diffusion-3.5-medium/feat-quantization`

**Pull request closed without merging**